src/hardware/gus.cpp

```diff
@@ -665,27 +673,37 @@ void Gus::ActivateVoices(uint8_t requested_voices)
 	}
 }
 
-AudioFrame Gus::RenderFrame()
+const std::vector<AudioFrame>& Gus::GetFrames(const int num_requested_frames)
 {
-	AudioFrame accumulator = {};
+	// use a static vector to avoid re-allocating each millisecond
+	static std::vector<AudioFrame> frames = {};
```

Comment:
How about storing this in the `Gus` class instead? Would be a bit cleaner that way, no?

Reply:
Yup.. will make it a member.